doc(helm): remove Tiller reference from the docs

hickeyma opened the pull request:
Partial #5397

hickeyma committed:
Remove Tiller reference from the docs

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

adamreese commented on docs/chart_template_guide/builtin_objects.md:
```diff
@@ -21,7 +20,6 @@ In the previous section, we use `{{.Release.Name}}` to insert the name of a rele
   - `Capabilities.APIVersions` is a set of versions.
   - `Capabilities.APIVersions.Has $version` indicates whether a version (`batch/v1`) is enabled on the cluster.
   - `Capabilities.KubeVersion` provides a way to look up the Kubernetes version. It has the following values: `Major`, `Minor`, `GitVersion`, `GitCommit`, `GitTreeState`, `BuildDate`, `GoVersion`, `Compiler`, and `Platform`.
-  - `Capabilities.helmVersion` provides a way to look up the Tiller version. It has the following values: `SemVer`, `GitCommit`, and `GitTreeState`.
```
This line is still valid. s/Tiller/Helm/

adamreese commented on docs/architecture.md:
```diff
@@ -24,41 +24,33 @@ For Helm, there are three important concepts:
-The Tiller server stores information in ConfigMaps located inside of
-Kubernetes. It does not need its own database.
+The library uses the Kubernetes client library to communicate with Kubernetes. Currently,
+that library uses REST+JSON. Information in ConfigMaps located inside of Kubernetes.
```
```suggestion
that library uses REST+JSON. Information in Secrets located inside of Kubernetes.
```

adamreese commented on docs/chart_best_practices/conventions.md:
```diff
@@ -28,32 +28,12 @@ When SemVer versions are stored in Kubernetes labels, we conventionally alter th
 
 When in doubt, use _Helm_ (with an uppercase 'H').
-
-## Restricting Tiller by Version
```
This section is still valid. s/Tiller/Helm/

adamreese commented on docs/charts.md:
```diff
@@ -487,8 +486,7 @@ the Kubernetes objects from the charts and all its dependencies are
 
-The install order of Kubernetes types is given by the enumeration InstallOrder in kind_sorter.go
-(see [the Helm source file](https://github.com/kubernetes/helm/blob/master/pkg/tiller/kind_sorter.go#L26)).
+The install order of Kubernetes types is given by the enumeration InstallOrder in ***TBD***
```
you can use the v3 branch url https://github.com/helm/helm/blob/dev-v3/pkg/tiller/kind_sorter.go#L26

adamreese commented on docs/charts.md:
```diff
@@ -589,9 +586,8 @@ sensitive_.
   as `[]byte` using `{{.Files.GetBytes}}`
 - `Capabilities`: A map-like object that contains information about the versions
-  of Kubernetes (`{{.Capabilities.KubeVersion}}`, Tiller
-  (`{{.Capabilities.HelmVersion}}`, and the supported Kubernetes API versions
```
Keep `{{.Capabilities.HelmVersion}}`

adamreese commented on docs/install.md:
```diff
@@ -48,7 +45,7 @@ choco install kubernetes-helm
 
 Helm now has an installer script that will automatically grab the latest version
-of the Helm client and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
+of the Helm and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
```
```suggestion
of Helm and [install it locally](https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get).
```

adamreese reviewed (changes requested):
@hickeyma Thanks for this!  Just a few notes.

hickeyma committed:
Update comments after review

- https://github.com/helm/helm/pull/4788#discussion_r226037034
- https://github.com/helm/helm/pull/4788#discussion_r226037064
- https://github.com/helm/helm/pull/4788#discussion_r226037806
- https://github.com/helm/helm/pull/4788#discussion_r226038492
- https://github.com/helm/helm/pull/4788#discussion_r226039202
- https://github.com/helm/helm/pull/4788#discussion_r226039894

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Thanks for merging @adamreese
